application-admintools-test/application-admintools-test-docker/src/test/it/com/xwiki/admintools/test/ui/AdminToolsIT.java

```diff
@@ -0,0 +1,300 @@
+        }
+
+        // By default the minimal distribution used for the tests doesn't have any rights setup. Let's create an Admin
+        // user part of the Admin Group and make sure that this Admin Group has admin rights in the wiki. We could also
+        // have given that Admin user the admin right directly but the solution we chose is closer to the XS
+        // distribution.
+        setup.loginAsSuperAdmin();
+        setup.setGlobalRights("XWiki.XWikiAdminGroup", "", "admin", true);
+        setup.createAdminUser();
+        setup.loginAsAdmin();
+    }
+
+    @BeforeEach
+    void goToPage(TestUtils testUtils)
+    {
+        AdminToolsHomePage page = AdminToolsHomePage.gotoPage();
+        page.waitUntilPageIsReady();
+    }
+
+    @Test
+    void appEntryRedirectsToHomePage()
+    {
+        ApplicationsPanel applicationPanel = ApplicationsPanel.gotoPage();
+        ViewPage vp = applicationPanel.clickApplication("Admin Tools");
+        Assertions.assertEquals(AdminToolsHomePage.getSpace(), vp.getMetaDataValue("space"));
+        assertEquals(AdminToolsHomePage.getPage(), vp.getMetaDataValue("page"));
+    }
+
+    @Test
+    @Order(1)
+    void adminToolsHomePageBackend(TestConfiguration testConfiguration)
+    {
+        if (!isSupportedServer) {
+            return;
+        }
+        AdminToolsViewPage webHomePage = new AdminToolsViewPage();
+        assertEquals(2, webHomePage.getDashboardElements().size());
+        String backendText = webHomePage.getBackendText();
+        assertTrue(supportedServers.stream().anyMatch(s -> backendText.toLowerCase().contains(s.toLowerCase())));
+
+        String configurationDatabase = testConfiguration.getDatabase().name().toLowerCase();
+        List<WebElement> warningMessages = webHomePage.getBackendErrorMessages();
+        if (supportedDatabases.stream().anyMatch(configurationDatabase::contains)) {
+            assertEquals(0, warningMessages.size());
+        } else {
+            assertEquals(1, warningMessages.size());
+        }
+        assertTrue(supportedDatabases.stream().anyMatch(d -> backendText.toLowerCase().contains(d)));
+        assertFalse(backendText.toLowerCase().contains("null"));
+    }
+
+    @Test
+    @Order(2)
+    void adminToolViewLastLogLinesModal(TestUtils testUtils)
+    {
+        if (!isSupportedServer) {
+            return;
+        }
+        AdminToolsViewPage webHomePage = new AdminToolsViewPage();
+        WebElement viewLastLogsHyperlink = webHomePage.getBackendLogsHyperlink();
+        viewLastLogsHyperlink.click();
+        LastNLinesModalView lastLinesModalView = new LastNLinesModalView();
+        lastLinesModalView.waitUntilPageIsReady();
+        String mainWindowHandle = testUtils.getDriver().getWindowHandle();
+        lastLinesModalView.clickButton(BACKEND_SECTION_VIEW_LAST_LOGS_MODAL_ID);
+        switchToNewTab(testUtils, mainWindowHandle);
+        testUtils.getDriver().switchTo().window(mainWindowHandle);
```

Comment:
Why do you need to switch tabs?

Reply:
I need to manually switch tabs to verify the opened content, as Selenium does not change focus when a tab is opened. I had a [comment](https://github.com/xwikisas/application-admintools/pull/25/files#diff-9a24a56c8d4cfc69ff20d4fbab0164f8e55c3304f76af2117973119a5f56ded9R260-R262) explaining this but did not offer a lot of details.